mbf_mesh_nav/src/mesh_navigation_server.cpp

```diff
@@ -37,13 +37,94 @@
 
 #include "mbf_mesh_nav/mesh_navigation_server.h"
 
+#include <algorithm>
 #include <functional>
 
 #include <geometry_msgs/msg/pose_array.hpp>
 #include <mesh_map/mesh_map.h>
 #include <nav_msgs/msg/path.hpp>
 #include <rclcpp/logging.hpp>
 
+namespace {
+//! Helper function, intended for formatting available plugin types.
+//! Returns a string like this: [el1, el2, el3, ..., eln]
+std::string stringVectorToString(const std::vector<std::string>& vec) 
+{
+  std::stringstream s;
+  s << "[";
+  bool is_first = true;
+  for (const auto& item : vec)
+  {
+    s << (is_first ? "" : ", ") << item;
+    is_first = false;
+  }
+  s << "]";
+  return s.str();
+};
+
+/*!
+ * Helper function to reduce code duplication when loading plugins.
+ * Things can be either planners, controllers or recovery behaviors.
+ * @tparam AbstractThing Abstract type of the thing to load. Needs to be the base class of SimpleThing and MeshThing. E.g. mbf_abstract_core::AbstractPlanner.
+ * @tparam SimpleThing Simple type of the thing to load. E.g. mbf_simple_core::SimplePlanner
+ * @tparam MeshThing Mesh type of the thing to load. E.g. mbf_simple_core::MeshPlanner
+ * @param thing_type typename of the thing that shall be loaded, usually configured by user via ros params.
+ * @param which_thing should be "planner", "controller", or "recovery behavior", used only for logging.
+ */
+template<typename AbstractThing, typename MeshThing, typename SimpleThing>
```

Comment:
I would call these template and "normal" parameters Plugins rather than thing since the function is also called "loadPlugin" rather than "loadThing". It took me some time to understand what the different "Things" are in this function.

Reply:
Hm, yes, `thing` is a very nonspecific name. I'd love to use something else.
I think I considered `{Abstract,Mesh,Simple}PluginType` as template args. but I was afraid of the naming conflict with with the `const std::string& plugin_type` parameter.
But now this param is called `thing_type` anyways, so it's in not at all better.

Maybe `{Abstract,Mesh,Simple}CorePluginType`, so specify that these template args need to be one of the core types and not a specific type like `StanleyController`. `"stanley_controller"` would fit into the normal parameter `plugin_type`, though.